pkg/storage/unified/apistore/prepare.go

```diff
@@ -131,8 +136,33 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 		obj.SetDeprecatedInternalID(previousInternalID) // nolint:staticcheck
 	}
 
-	obj.SetUpdatedBy(info.GetUID())
-	obj.SetUpdatedTimestampMillis(time.Now().UnixMilli())
+	// Check if we should bump the generation
+	changed := obj.GetFolder() != previous.GetFolder()
+	if changed {
+		if !s.opts.EnableFolderSupport {
+			return nil, apierrors.NewBadRequest(fmt.Sprintf("folders are not supported for: %s", s.gr.String()))
+		}
+		// TODO: check that we can move the folder?
+	} else {
+		spec, e1 := obj.GetSpec()
+		oldSpec, e2 := previous.GetSpec()
+		if e1 == nil && e2 == nil {
+			if !apiequality.Semantic.DeepEqual(spec, oldSpec) {
+				changed = true
+			}
+		}
+	}
+
+	// Mark the resource as changed
+	if changed {
+		obj.SetGeneration(previous.GetGeneration() + 1)
+		obj.SetUpdatedBy(info.GetUID())
+		obj.SetUpdatedTimestampMillis(time.Now().UnixMilli())
```

Comment:
@chaudyg hopefully this will help your comment https://github.com/grafana/grafana/pull/102528